__tests__/unit/git.test.ts

```diff
@@ -0,0 +1,69 @@
+import {createTempEmptyDir} from '../../src/__tests__/helpers'
+
+import {Git} from '../../src/git'
+import {GitRepoDir} from '../../src/git-repo-dir'
+
+function gitEnv(): NodeJS.ProcessEnv {
+  return {
+    LANG: 'en_US.UTF-8'
+  }
+}
+
+async function newGitInstanceInEmptyWorkingDir(): Promise<Git> {
+  const gitRepoDir = new GitRepoDir(await createTempEmptyDir())
+  const git = new Git(gitRepoDir, gitEnv())
+  return git
+}
+
+describe('Git', () => {
+  it('should fail running `git status` when the repo has not been initialized', async () => {
+    const git = await newGitInstanceInEmptyWorkingDir()
+
+    const checkStatus = (): string | Buffer => {
+      return git.status()
+    }
+
+    expect(checkStatus).toThrowError()
+  })
+
+  it('should not fail running `git status` when the repo has been initialized', async () => {
+    const git = await newGitInstanceInEmptyWorkingDir()
+
+    git.init()
+
+    expect(git.status()).toMatch('On branch')
+  })
+
+  it('should fail running `git log` when the repo has not been initialized', async () => {
+    const git = await newGitInstanceInEmptyWorkingDir()
+
+    const checkStatus = (): string | Buffer => {
+      return git.log()
+    }
+
+    expect(checkStatus).toThrowError()
+  })
+
+  it('should fail running `git log` when the repo has been initialized but it does not have a ny commits', async () => {
+    const git = await newGitInstanceInEmptyWorkingDir()
+
+    git.init()
+
+    const checkStatus = (): string | Buffer => {
+      return git.log()
+    }
+
+    expect(checkStatus).toThrowError()
+  })
+
+  it('should not fail running `git log` when the repo has been initialized and it has at least one commit', async () => {
+    const git = await newGitInstanceInEmptyWorkingDir()
+
+    git.init()
+    git.setLocalConfig('user.name', 'A committer')
+    git.setLocalConfig('user.email', 'committer@example.com')
+    git.emptyCommit('not relevant')
```

Comment:
I'm not sure about it, but here we are testing two inter-dependent things in the same test: that 'git log' do not fail when there is a commit and that the emptyCommit actually creates the commit.
What if emptyCommit creates the commit but not as we want? Maybe we should test it independently.

Reply:
A very nice approach that last one. Thanks for the clarifications.

Reply:
hi @yeraydavidrodriguez in the end I realized that Simple-Git has a `raw` method to execute custom git commands. I'm using that method instead of adding a new Git wrapper.

Changes are implemented in this PR https://github.com/Nautilus-Cyberneering/git-queue/pull/259

- [Here](https://github.com/josecelano/git-queue/blob/issue-198-fix-code-scanning-alert/src/git-repo.ts#L18)
- [And here](https://github.com/josecelano/git-queue/blob/issue-198-fix-code-scanning-alert/src/git-repo.ts#L48)

Regarding the GitStatusBuidler we do not have any complex scenario right now. FOr now, I'm also using the `raw` method. See [here](https://github.com/josecelano/git-queue/blob/issue-198-fix-code-scanning-alert/__tests__/unit/git-repo.test.ts#L43-L61).

This is the sample test where I need to create a new commit:

```typescript
  it('should check if an initialized repo has commits', async () => {
    const gitRepoDir = new GitRepoDir(await createTempEmptyDir())
    const git = await newSimpleGitWithCommitterIdentity(gitRepoDir)

    const gitRepo = new GitRepo(gitRepoDir, git)
    await gitRepo.init()

    expect(await gitRepo.hasCommits()).toBe(false)

    await git.raw(
      'commit',
      '--no-gpg-sign',
      '--allow-empty',
      '--message',
      'Initial commit'
    )

    expect(await gitRepo.hasCommits()).toBe(true)
  })
```

If we need to create more complex git repo states for testing we can create the `GitStatusBuidler` but using also [Simple-Git](https://github.com/steveukx/git-js#readme).

Reply:
Great, @josecelano , it seems a good choice to avoid adding unnecessary complexity layers.